docs/source/reference/modding/extended.rst

```diff
@@ -0,0 +1,211 @@
+OpenMW transfers the contents directly into the player's inventory, triggers a theft crime event if the plant is owned and toggles the harvested model state.
+
+It is also possible to use scripts with ``OnActivate`` command for such containers. For example, when player needs a tool to harvest a plant (e.g. a pickaxe for ore).
+
+Keep in mind that the collision detection system ignores switch nodes, so add a ``RootCollisionNode`` or ``NCO`` NiStringExtraData to harvestable meshes.
+
+Advantages of described approach over old herbalism mods:
+
+1. There is no need to spawn separate "harvested" objects
+
+2. There is no need to attach a script to every container
+
+3. It supports an ownership check (the original engine without MWSE does not)
+
+4. It does not alter an original respawn mechanics
+
+An example of mod which uses this feature is `Graphic Herbalism`_.
+
+Animated containers support
+---------------------------
+
+It is possible to attach opening/closing animations for containers. To do it, you need to create a KF-file for container with such groups:
+
+1. ``Idle`` (with ``Start`` and ``Stop`` keys)
+
+2. ``ContainerOpen`` (with ``Start``, ``Loot`` and ``Stop`` keys)
+
+3. ``ContainerClose`` (with ``Start`` and ``Stop`` keys)
+
+The ``Idle`` group can have the zero time, the ``Loot`` key for ``ContainerOpen`` allows to play a part of opening animation in the background.
+
+For example, with the following setup, the opening animation has 1.0 sec duration and shows the container window in the middle of opening animation:
+
+::
+
+    0.0: ContainerOpen: start
+    0.5: ContainerOpen: loot
+    1.0: ContainerOpen: stop
+
+It is also possible to attach opening/closing sounds to container's animations:
+
+::
+
+    1.0: ContainerClose: open
+    1.01: Sound: AC_dw_drawer_close
+    2.0: ContainerClose: stop
+
+An approach is the same as for other animation sounds (you need to register sound record in the ESM/ESP file).
+Note that the sound starting time value is slightly higher than the closing animation start, otherwise sound will be played at the end of opening animation in this example.
+
+It is important to assign a RootCollisionNode to the container mesh -- the collision shape will depend on the animation state otherwise, what can have a performance impact.
+
+Advantages of described approach over old animated containers mods:
+
+1. There is no need to attach a script to every container
+
+2. Part of opening animation can be played in the background, so we do not waste player's time
+
+An example of mod which uses this feature is `OpenMW Containers Animated`_.
+
+
+Day/night state support
+-----------------------
+
+It is possible to add one or more NiSwitchNodes named ``NightDaySwitch``.
+Every such switch should have at least two child nodes
+(first one represents the node in the normal mode, second one - in the night mode, optional third node - a day mode for interiors).
```

Comment:
(first one represents the node in the normal mode, second one - in the night mode, optional third node - a day mode for interiors).
```suggestion
(the first node represents the normal node state, the second one represents the node state during night, the optional third node represents the node state during daytime in interior cells).
```